### PR TITLE
Fix being unable to fit on a plot with no legend

### LINKED ISSUE #26974: Workbench: Cannot fit on plots with no legend
### Expected behavior
You shouldn't need a legend on your plot to do a fit.

### Actual behavior
The following error is raised:
```
Traceback (most recent call last):
  File "\mantid\qt\python\mantidqt\widgets\fitpropertybrowser\fitpropertybrowser.py", line 352, in fitting_done_slot
    props = LegendProperties.from_legend(axes.legend_)
  File "mantid\mantid\qt\python\mantidqt\widgets\plotconfigdialog\legendtabwidget\__init__.py", line 26, in from_legend
    props['visible'] = legend.get_visible()
AttributeError: 'NoneType' object has no attribute 'get_visible'
```

### Steps to reproduce the behavior
- Run the following in workbench:
```python
from mantid.simpleapi import *

import math
import matplotlib.pyplot as plt
import numpy as np
from scipy import stats

np.random.seed(101)

mu = 0
variance = 1
sigma = math.sqrt(variance)
x = np.linspace(mu - 3*sigma, mu + 3*sigma, 100)
y = stats.norm.pdf(x, mu, sigma) + np.random.randn(100)*0.005
err = 0.01*np.ones(100)

ws = CreateWorkspace(x, y, err, OutputWorkspace='ws')
fig, ax = plt.subplots(1, 1, subplot_kw={'projection': 'mantid'})
ax.errorbar(ws, specNum=1)
plt.show()
```
- Open the fit browser, add a Gaussian peak and attempt to fit.
### Platforms affected
All

### PR DESCRIPTION
**Pull request by PhilColebrooke**
**Description of work.**
This PR fixes an issue where attempting to fit on a plot with no legend throws an error. It also makes it so that doing a fit or plotting a guess on a plot with no legend does not create a legend.

**To test:**
1. Run the following script in Workbench:
```
from mantid.simpleapi import *

import math
import matplotlib.pyplot as plt
import numpy as np
from scipy import stats

np.random.seed(101)

mu = 0
variance = 1
sigma = math.sqrt(variance)
x = np.linspace(mu - 3*sigma, mu + 3*sigma, 100)
y = stats.norm.pdf(x, mu, sigma) + np.random.randn(100)*0.005
err = 0.01*np.ones(100)

ws = CreateWorkspace(x, y, err, OutputWorkspace='ws')
fig, ax = plt.subplots(1, 1, subplot_kw={'projection': 'mantid'})
ax.errorbar(ws, specNum=1)
plt.show()
```

Plot a guess and fit and see that no error is thrown and a legend is not created.

Fixes #26974 

No release notes needed because this is bug isn't present in a release.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowser.py [hunk 1]
```diff
@@ -368,10 +377,14 @@ def fitting_done_slot(self, name):
         for new_line, old_line in zip(new_lines, original_lines):
             new_line.update_from(old_line)
 
-        handles, labels = axes.get_legend_handles_labels()
+        if plot_has_legend:
+            handles, labels = axes.get_legend_handles_labels()
```
Comment (Harrietbrown) [inline qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowser.py]: What does this function do, `handles` and `labels` do not seem to be used.

**Review by Harrietbrown — COMMENTED** [review]
Good work. It is stable and is able to handle having another workspace overplotted while it is working, Just one question before going forward.

File: qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowser.py (hunk 1 above)
Comment (PhilColebrooke) [reply qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowser.py]: I checked with the author and it was left in by mistake after resolving a merge conflict. Will remove 👍

**Review by Harrietbrown — DISMISSED** [review]
Excelent Work. I was glad to see it able to handle new lines with a legend added while still in fitting mode.

**Comment by gvardany** [comment]
This has developed a merge conflict.

**Comment by PhilColebrooke** [comment]
@Harrietbrown I had to fix some things to account for a recent merge so would you mind having another check over this to make sure everything still works?

**Review by Harrietbrown — APPROVED** [review]
Again, this is good work, every still works, once the tests have finished it should be good to merge.

**Pull request merged**